src/pages/iou/steps/NewRequestAmountPage.js

```diff
@@ -0,0 +1,208 @@
+import * as ReportUtils from '../../../libs/ReportUtils';
+import * as CurrencyUtils from '../../../libs/CurrencyUtils';
+import CONST from '../../../CONST';
+import reportPropTypes from '../../reportPropTypes';
+import * as IOU from '../../../libs/actions/IOU';
+import useLocalize from '../../../hooks/useLocalize';
+import MoneyRequestAmountForm from './MoneyRequestAmountForm';
+import * as IOUUtils from '../../../libs/IOUUtils';
+import FullPageNotFoundView from '../../../components/BlockingViews/FullPageNotFoundView';
+import styles from '../../../styles/styles';
+import HeaderWithBackButton from '../../../components/HeaderWithBackButton';
+import ScreenWrapper from '../../../components/ScreenWrapper';
+
+const propTypes = {
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            iouType: PropTypes.string,
+            reportID: PropTypes.string,
+        }),
+    }),
+
+    /** The report on which the request is initiated on */
+    report: reportPropTypes,
+
+    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+    iou: PropTypes.shape({
+        id: PropTypes.string,
+        amount: PropTypes.number,
+        currency: PropTypes.string,
+        participants: PropTypes.arrayOf(
+            PropTypes.shape({
+                accountID: PropTypes.number,
+                login: PropTypes.string,
+                isPolicyExpenseChat: PropTypes.bool,
+                isOwnPolicyExpenseChat: PropTypes.bool,
+                selected: PropTypes.bool,
+            }),
+        ),
+    }),
+
+    // eslint-disable-next-line react/forbid-prop-types
+    errors: PropTypes.object,
+};
+
+const defaultProps = {
+    route: {
+        params: {
+            iouType: '',
+            reportID: '',
+        },
+    },
+    report: {},
+    iou: {
+        id: '',
+        amount: 0,
+        currency: CONST.CURRENCY.USD,
+        participants: [],
+    },
+    errors: {},
+};
+
+function NewRequestAmountPage({route, iou, report, errors}) {
+    const {translate} = useLocalize();
+
+    const prevMoneyRequestID = useRef(iou.id);
+    const textInput = useRef(null);
+
+    const iouType = lodashGet(route, 'params.iouType', '');
+    const reportID = lodashGet(route, 'params.reportID', '');
+    const isEditing = lodashGet(route, 'path', '').includes('amount');
+    const currentCurrency = lodashGet(route, 'params.currency', '');
+
+    const currency = currentCurrency || iou.currency;
+
+    const focusTextInput = () => {
+        // Component may not be initialized due to navigation transitions
+        // Wait until interactions are complete before trying to focus
+        InteractionManager.runAfterInteractions(() => {
+            // Focus text input
+            if (!textInput.current) {
+                return;
+            }
+
+            textInput.current.focus();
+        });
+    };
+
+    useFocusEffect(
+        useCallback(() => {
+            focusTextInput();
```

Comment:
Coming from https://github.com/Expensify/App/issues/22388, this has caused a small regression 
We're focusing the input immediately after the page is focused. When this happens when the page starts navigating, it will cancel the transition animation and we just see an abrupt page change

It was later decided to use [this solution](https://expensify.slack.com/archives/C01GTK53T8Q/p1694660990479979) whenever we need to autofocus a TexInput, which doesn't have this problem

Reply:
@eVoloshchak It was already implemented earlier https://github.com/Expensify/App/blob/e2694f52ce61fa9e67ee7ab1636cc38b2ba24d48/src/pages/iou/steps/MoneyRequestAmount.js#L292-L296. It was just moved to the new file 🙂

Reply:
Oops, I see, thank you!